lib/node_loader.js

```diff
@@ -95,6 +96,7 @@ AWS.CredentialProviderChain.defaultProviders = [
   function () { return new AWS.EnvironmentCredentials('AMAZON'); },
   function () { return new AWS.SharedIniFileCredentials(); },
   function () { return new AWS.ECSCredentials(); },
+  function () { return new AWS.SsoCredentials(); },
```

Comment:
Sso credentials should get preference over Shared Ini credentials.
```suggestion
  function () { return new AWS.SsoCredentials(); },
  function () { return new AWS.SharedIniFileCredentials(); },
  function () { return new AWS.ECSCredentials(); },
```

Equivalent code in v3: https://github.com/aws/aws-sdk-js-v3/blob/11c4a7b5ac00763139f130595a37e182d34d9a72/packages/credential-provider-node/src/defaultProvider.ts#L57-L60